### PR TITLE
FUSETOOLS-2589: UI Tests - Extend 'DownloadServerTest'

### PR DESCRIPTION
**Pull request by pospisilf**
# Pull Request Checklist

After this checklist is all checked or PR provides explanations for possible pass-through, please put the label "Ready for review" on the PR.

## General

- [ ] Did you use the Jira Issue number in the commit comments?
- [ ] Did you set meaningful commit comments on each commit?
- [ ] Did you sign off all commits for this PR? (git commit -s -m "jira issue number - your commit comment")

## Functional

- [ ] Did the CI job report a successful build?
- [ ] Is the non-happy path working, too?

## Maintainability

- [ ] Are all Sonar reported issues fixed or can they be ignored?
- [ ] Is there no duplicated code?
- [ ] Are method-/class-/variable-names meaningful?

## Tests

- [ ] Are there unit-tests?
- [ ] Are there integration tests (or at least a jira to tackle these)?
- [ ] Do we need a new UI test?

## Legal

- [ ] Have you used the correct file header copyright comment?
- [ ] Have you used the correct year in the headers of new files?

### CODE REVIEW

File: uitests/tests/org.jboss.tools.fuse.ui.bot.tests/src/org/jboss/tools/fuse/ui/bot/tests/DownloadServerTest.java [hunk 1]
```diff
@@ -71,58 +114,90 @@
...
+		 */
 		new LabeledText(DOWNLOAD_FOLDER).setText(ResourceHelper.getResourceAbsolutePath(Activator.PLUGIN_ID, "target"));
 		new LabeledText(INSTALL_FOLDER).setText(ResourceHelper.getResourceAbsolutePath(Activator.PLUGIN_ID, "target"));
-		buttonFinish.click();
+		new PushButton(FINISH_BUTTON).click();
 
 		AbstractWait.sleep(TimePeriod.getCustom(2));
 		new WaitWhile(new JobIsRunning(), TimePeriod.getCustom(3600));
 		AbstractWait.sleep(TimePeriod.getCustom(2));
+		new WaitUntil(new ButtonIsAvailable(FINISH_BUTTON));
```
[the remark this answers is not in the document]
Comment (pospisilf) [reply uitests/tests/org.jboss.tools.fuse.ui.bot.tests/src/org/jboss/tools/fuse/ui/bot/tests/DownloadServerTest.java:200]: Yes, in both cases you mentioned will this solution work. This buttons must be sooner or later enabled to continue in test and finish with success. 'ControlIsEnabled' returns 'true' when button is available and enabled. On line 163 is case of button, which is not always enabled. Test is adding credentials into login form, then OK button is available and enabled. But on next run is not possible to add same credentials again, OK button will be available but not enabled. 'ControlIsEnabled' returns false, 'ButtonIsAvailable' returns true.

**Comment by djelinek** [comment]
testPR

**Review by djelinek — APPROVED** [review]

File: uitests/tests/org.jboss.tools.fuse.ui.bot.tests/src/org/jboss/tools/fuse/ui/bot/tests/DownloadServerTest.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (apupier) [reply uitests/tests/org.jboss.tools.fuse.ui.bot.tests/src/org/jboss/tools/fuse/ui/bot/tests/DownloadServerTest.java:200]: I do not understand the subtility :-(

right after the button is clicked so we need to wait the button to be enabled and not only available.

File: uitests/tests/org.jboss.tools.fuse.ui.bot.tests/src/org/jboss/tools/fuse/ui/bot/tests/DownloadServerTest.java [hunk 2]
```diff
@@ -71,58 +114,90 @@
...
+			AbstractWait.sleep(TimePeriod.getCustom(2));
+			new WaitUntil(new ButtonIsAvailable(NEXT_BUTTON));
+			assertTrue(buttonNext.isEnabled());
+			buttonNext.click();
+
+			new WaitUntil(new RadioButtonIsAvailable(ACCEPT_TERMS));
+		}
 
+		/*
+		 * Accept licence agreement.
```
Comment (apupier) [inline uitests/tests/org.jboss.tools.fuse.ui.bot.tests/src/org/jboss/tools/fuse/ui/bot/tests/DownloadServerTest.java:184]: licence -> license

**Comment by pospisilf** [comment]
Will be redone in different way.

**Pull request closed without merging**